datalad/interface/save.py

```diff
@@ -81,6 +83,26 @@ def __call__(message=None, files=None, dataset=None,
         if auto_add_changes:
             files = [ds.path]
 
+        # treat special case of still untracked subdatasets.
+        # those need to become submodules now, as they are otherwise added
+        # without an entry in .gitmodules, and subsequently break Git's
+        # submodule functionality completely
+        for utf in ds.repo.repo.untracked_files:
+            utf_abspath = opj(ds.path, utf)
+            if not isdir(utf_abspath):
+                # this cannot be a repository
+                continue
+
+            # test whether the potential submodule is scheduled for saving
+            if sum([realpath(utf_abspath).startswith(_with_sep(realpath(f)))
+                    for f in files]):
```

Comment:
OPT: `_with_sep(realpath(f))` would be called on the same set of files for every untracked file -- so why not to pre-compute all those to save at least some cycles?